### PR TITLE
etcd.py module improved . added posibility to configure from playbook…

### PR DESCRIPTION
**Pull request by MihaiMoldovanu**
##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

plugin: lookup/etcd.py

##### ANSIBLE VERSION

```
ansible 2.0.0.2
```

##### SUMMARY

Improved lookup/etcd.py added posibity to check remote etcd servers set directly from playbooks

```

```

… etcd server to query

gundalow changed this to a feature (rather than bug) pull request

### CODE REVIEW

File: docsite/rst/playbooks_lookups.rst [hunk 1]
```diff
@@ -368,6 +368,29 @@ TLSA         usage, selector, mtype, cert
...
+
+The etcd Lookup
+````````````````````
+.. versionadded:: 2.3
+
+When you need to query a value from an etcd server you can use etcd lookup. Until version 2.3 you could interrogate localmachine by default or using environment variable you could interrogate a remote v1 server. Since 2.3 this functionality has been improved by allowing to set the etcd host and version directly from you playbook:
+
+Example usage:
+
+# query localhost etcd ( v1 )
```
Comment (gundalow) [inline docsite/rst/playbooks_lookups.rst]: If you indent this sample block, like in "Here are some examples" under "More Lookups" that should fix the formatting

File: docsite/rst/playbooks_lookups.rst [hunk 2]
```diff
@@ -368,6 +368,29 @@ TLSA         usage, selector, mtype, cert
...
+- hosts: localhost
+  connection: local
+  tasks:
+   - debug: msg="{{ lookup('etcd', 'foo') }} is a value from a locally running etcd"
+
+# query a remote v2 etcd server
+
```
Comment (gundalow) [inline docsite/rst/playbooks_lookups.rst]: # Note "url" option is new in Ansible 2.3

**Review by gundalow — CHANGES_REQUESTED** [review]

**Comment by gundalow** [comment]
Do you think the parameters should be in a table, like "The CSV File Lookup"

File: docsite/rst/playbooks_lookups.rst [hunk 3]
```diff
@@ -368,6 +368,42 @@ TLSA         usage, selector, mtype, cert
...
+When you need to query a value from an etcd server you can use etcd lookup. 
+Before version 2.3 you could interrogate localmachine by default or using environment variable you could interrogate a remote v1 server.
+
+Since 2.3 this functionality has been improved by allowing to set the etcd host and version directly from you playbook
+
+==============   =============================================================================
+Parameters       Values
+--------------   -----------------------------------------------------------------------------
+url              Default 'http://127.0.0.1:4001'. If ANSIBLE_ETCD_URL is set will override
+version          Default 'v1'. If ANSIBLE_ETCD_VERSION is set will override default
```
Comment (gundalow) [inline docsite/rst/playbooks_lookups.rst]: ". Added in 2.3"

File: docsite/rst/playbooks_lookups.rst [hunk 4]
```diff
@@ -368,6 +368,42 @@ TLSA         usage, selector, mtype, cert
...
+
+When you need to query a value from an etcd server you can use etcd lookup. 
+Before version 2.3 you could interrogate localmachine by default or using environment variable you could interrogate a remote v1 server.
+
+Since 2.3 this functionality has been improved by allowing to set the etcd host and version directly from you playbook
+
+==============   =============================================================================
+Parameters       Values
+--------------   -----------------------------------------------------------------------------
+url              Default 'http://127.0.0.1:4001'. If ANSIBLE_ETCD_URL is set will override
```
Comment (gundalow) [inline docsite/rst/playbooks_lookups.rst]: ". Added in 2.3"

**Review by gundalow — APPROVED** [review]
shipit

**Comment by gundalow** [comment]
Once merged we should add this to `CHANGELOG`

**Comment by calfonso** [comment]
@MihaiMoldovanu could you rebase this and resolve any conflicts, and then we'll get it reviewed/merged. Thanks!

**Comment by MihaiMoldovanu** [comment]
Ok :) Done :) @calfonso anything else i should do ?

**Comment by MihaiMoldovanu** [comment]
Fixed the conflict

**Comment by MihaiMoldovanu** [comment]
Everything in green . Can you merge it  ?

**Comment by gundalow** [comment]
@MihaiMoldovanu Rebase needed, see the details in the previous `ansibot` comment

**Comment by MihaiMoldovanu** [comment]
Rebased .

**Comment by MihaiMoldovanu** [comment]
@gundalow  now it's ok ?

File: docs/docsite/rst/playbooks_lookups.rst [hunk 5]
```diff
@@ -508,6 +508,40 @@ Sample output:
...
+Variables set in playbook will override defaults.
+
+Example usage::
+
+    # query localhost etcd ( v1 )
+    - hosts: localhost
+      connection: local
+      tasks:
+       - debug: msg="{{ lookup('etcd', 'foo') }} is a value from a locally running etcd"
+
```
Comment (gundalow) [inline docs/docsite/rst/playbooks_lookups.rst]: Multiline please

```
- debug:
    msg: "{{ lookup('etcd', 'foo') }} is a value from a locally running etcd"
```

File: docs/docsite/rst/playbooks_lookups.rst [hunk 6]
```diff
@@ -508,6 +508,40 @@ Sample output:
...
+    # query a remote v2 etcd server
+    # Note "url" option is new in Ansible 2.5
+    - hosts: localhost
+      connection: local
+       - debug: msg={{lookup('etcd','/tfm/network/config', url='http://192.168.1.21:2379' , version='v2')}}
```
Comment (gundalow) [inline docs/docsite/rst/playbooks_lookups.rst]: Multiline format please

**Review by gundalow — APPROVED** [review]

**Review by dharmabumstead — APPROVED** [review]
Approved with some edits. Thanks @MihaiMoldovanu!

**Comment by MihaiMoldovanu** [comment]
Erm ... playbook_lookups changed a lot . lookup options / examples were removed from playbook_lookups.rst .

**Comment by bcoca** [comment]
@MihaiMoldovanu lookup docs now live in the lookup itself

**Comment by MihaiMoldovanu** [comment]
fixed conflict

**Comment by gundalow** [comment]
Merged into `devel`, will be released in `2.5.0`

**Pull request merged**